content/articles/understanding-diffie-helman-and-elgamal-asymmetric-encryption/index.md

```diff
@@ -0,0 +1,181 @@
+---
+layout: engineering-education
+status: draft
+published: false
+url: /understanding-diffie-helman-and-elgamal-asymmetric-encryption/
+title: Understanding Diffie Helman and Elgamal Asymmetric Encryption
+description: This tutorial will help the reader understand the concept of Diffie Helman and Elgamal Asymmetric encryption algorithms together with key generation,encryption and decryption process.
+date: 2021-09-05T00:00:00-02:00 
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-diffie-helman-and-elgamal-asymmetric-encryption/hero.jpg
+    alt: Encryption Example Image
+---
+Security plays a crucial role in data communication and data protection. It helps prevent unauthorized access to confidential information which may lead to data loss and data modification by unknown people making data transferred safe and unchanged.
+<!--more-->
+This article will cover how one can understand various asymmetric encryption algorithms: Diffie Helman and Elgamal to protect data transferred from sender to receiver. It will equip the learner with the knowledge of how to encrypt and decrypt data send together with how to generate the encryption and decryption keys.
+
+### Table of contents
+   - [Prerequisite](#prerequisite)
+   - [Asymmetric Encryption.](#asymmetric-encryption)
+   - [Modular Exponential](#modular-exponential)
+   - [Modular Exponential Using Python](#modular-exponential-using-python)
+   - [Working of DHKE Protocol](#working-of-dhke-protocol)
+      - [steps involved](#steps-involved)
+   - [Application of DHKE in ElGamal](#application-of-dhke-in-elgamal)
+      - [Key Generation](#key-generation)
+      - [Encryption](#encryption)
+      - [Decryption](#decryption)
+   - [Conclusion](#conclusion)
+
+### Prerequisite
+To follow along, the reader should:
+- Have a basic understanding of security concepts.
+- Have a prior understanding of modular arithmetic.
+- Know how to use various development environments for writing and running python code.
+### Asymmetric Encryption.
+Is also known as public-key encryption.
+Encryption involves enciphering information in transit from authorized persons and deciphering by authorized persons using keys.
+Asymmetric encryption uses two keys:
+public-key- which is the encryption key and is known to everyone in the network.
+private key- which decryption key is known to only the receiver of the content.
+Asymmetric encryption is used to curb the problem facing symmetric encryption.
+this includes:
+1. Key distribution - Key distribution under symmetric encryption requires
```

Comment:
Formatting is off. Please work on using better ways to put across the information.